File: Unidad 3-Funciones/Ejercicio25.c
//Author: Aldebaran Lira
//Program that finds the corresponding ASCII-code
//Version 1.0

#include <stdio.h>

//Function prototypes
void readLetter(char *);
int findASCIIcode(int);
void printCode(int);

int main(){

  // Declaration of variables
  char letter;
  int ASCIIc;

  //Input: A letter of the alphabet
  readLetter(&letter);

  //Process: Find the ASCII-code
  ASCIIc=(letter);

  //Output: The ASCII-code
  printCode(ASCIIc);

  return 0;
}

void readLetter(char *_letter){
  scanf("%c",_letter); getchar();
}

int findASCIIcode(int _ASCIIc){
  return _ASCIIc;
}

void printCode(int _output){    
  printf("%d\n\n",_output);
}

File: Unidad 3-Funciones/Ejercicio25.py
'''
Author: Aldebaran Lira
Version 0.0
Find the ASCII-code
'''

#Functions

#Int
def readLetter():
    '''
    It reads a letter from the alphabet
        Args:
            _letter (str): Value of the variable
        Returns: 
            A letter of the alphabet
    '''
    _letter = str(input())
    
    return _letter

#Process
def findASCIIcode(_letter):
    '''
    It searches for the corresponding code
        Args:
            _letter (str): A letter of the alphabet
            _code (int): ASCII-code
        Returns:
            The correspondinf code
    '''
    _code=ord(_letter)
    
    return _code

#Output
def printResult(_code):
    '''
    It prints the corresponding code
        Args:
            _code (int):The corresponding code
        Returns:
            Nothing
    '''
    print(_code)
    
    
#Input
code=0
letter=0
letter=readLetter()

#Process
code=findASCIIcode(letter)

#Output
printResult(code)
